code/datums/brain_damage/psychological.dm

```diff
@@ -0,0 +1,330 @@
+		if(4)
+			owner.apply_status_effect(STATUS_EFFECT_SPASMS)
+		if(5)
+			ADD_TRAIT(owner, TRAIT_MUTE, PSYCH_TRAIT)
+
+
+/datum/brain_trauma/psychological/schizophrenia/delusional/proc/roll_bad_effect()
+	clear_bad_effect()
+	state = rand(0,5) //Upon getting 0 you will have a few minutes of silence. After that you are back on the ride!
+	addtimer(CALLBACK(src, .proc/roll_bad_effect), rand(1 MINUTES, 5 MINUTES))
+
+/datum/brain_trauma/psychological/schizophrenia/delusional/proc/clear_bad_effect()
+	switch(state)
+		if(1)
+			owner.cure_blind(PSYCH_TRAIT)
+		if(4)
+			owner.remove_status_effect(STATUS_EFFECT_SPASMS)
+		if(5)
+			REMOVE_TRAIT(owner, TRAIT_MUTE, PSYCH_TRAIT)
+
+/datum/brain_trauma/psychological/schizophrenia/delusional/on_gain()
+	roll_bad_effect()
+	ADD_TRAIT(owner, TRAIT_DELUSIONAL, PSYCH_TRAIT)
+	..()
+
+/datum/brain_trauma/psychological/schizophrenia/delusional/on_lose()
+	clear_bad_effect()
+	REMOVE_TRAIT(owner, TRAIT_DELUSIONAL, PSYCH_TRAIT)
+	..()
+
+
+/datum/brain_trauma/psychological/collector
+	name = "Obsessive collector disorder"
+	desc = "Mental disorder that is characterized by an intense need of collecting otherwise useless objects"
+	gain_text = "<span class='danger'>You feel an odd compulsion to objects</span>"
+	lose_text = "<span class='notice'>You feel free once again</span>"
+	scan_desc = "Electric imaging suggests a mental disorder is present"
+	cure_points = 1000
+	lithiated_mod = -2
+	control_mod = 4
+	fearless_mod = -1
+	relaxed_mod = 1
+	var/list/obj/item/possible_items = list(/obj/item/toy,/obj/item/stamp,/obj/item/shard,/obj/item/reagent_containers/food/drinks/bottle,list(/obj/item/bodypart/l_arm,/obj/item/bodypart/r_arm))
+	var/list/obj/item/needed_item
+	var/greed_level = 0
+
+/datum/brain_trauma/psychological/collector/on_life()
+	..()
+	if(!check_items())
+		SEND_SIGNAL(owner, COMSIG_ADD_MOOD_EVENT, "collectors_itch", /datum/mood_event/collectors_itch)
+		SEND_SIGNAL(owner, COMSIG_CLEAR_MOOD_EVENT, "collectors_satisfaction", /datum/mood_event/collectors_satisfacton)
+	else
+		SEND_SIGNAL(owner, COMSIG_CLEAR_MOOD_EVENT, "collectors_itch", /datum/mood_event/collectors_itch)
+		SEND_SIGNAL(owner, COMSIG_ADD_MOOD_EVENT, "collectors_satisfaction", /datum/mood_event/collectors_satisfacton)
+
+/datum/brain_trauma/psychological/collector/proc/check_items()
+	var/stolen_count
+	var/list/all_items = owner.GetAllContents()
+	for(var/obj/I in all_items) //Check for wanted items
+		if(is_type_in_typecache(I, needed_item))
+			stolen_count++
+	return stolen_count >= greed_level
+
+/datum/brain_trauma/psychological/collector/proc/increase_greed()
+	to_chat(owner, "<span class='boldwarning'>You feel  your greed expanding</span>")
+	greed_level++
+	addtimer(CALLBACK(src, .proc/increase_greed), 5 MINUTES)
+
+/datum/brain_trauma/psychological/collector/proc/announce_item()
+	switch(needed_item)
+		if(/obj/item/toy)
+			to_chat(owner, "<span class='boldwarning'>You feel a great need to collect toys of all kind!</span>")
```

Comment:
Better off to make `possible_items` an associated list and store the descriptions there